.changeset/tall-comics-compare.md

```diff
@@ -0,0 +1,5 @@
+---
+"@itwin/presentation-hierarchies": minor
+---
+
+Increased the speed of hierarchy filtering with large number of filtered paths.
```

Comment:
should list all the api additions / changes you made in this pr

Reply:
Added

Reply:
I'm now thinking it would be nice to also specify the scope of improvement, e.g.:

```
Increased the speed of hierarchy filtering with large number of filtered paths.

| Amount of paths | Before the change | After the change |
| --- | --- | --- |
| 1k | ...
| 10k | ...
| 50k | ...

In addition, changed `NodeParser` (return type of `HierarchyDefinition.parseNode`):
- It now can return a promise, so instead of just `SourceInstanceHierarchyNode` it can now also return `Promise<SourceInstanceHierarchyNode>`.
- Additionally, it now accepts an optional `parentNode` argument of `HierarchyDefinitionParentNode` type.
```

Reply:
Updated